transformer_engine/pytorch/csrc/userbuffers/userbuffers.h

```diff
@@ -14,6 +14,13 @@
 #include <chrono>
 #include <stdexcept>
 
+#if defined(NVTE_WITH_MNNVL) && CUDA_VERSION >= 12040
+#include "nvml.h"
+#define MNNVL 1
+#else
+#define MNNVL 0
```

Comment:
This header is imported outside of UB, so a macro name like `NVTE_WITH_NVML` or `UB_WITH_NVML` is less likely to experience a name conflict:
```suggestion
#define UB_WITH_NVML 1
#else
#define UB_WITH_NVML 0
```